### ISSUE TITLE
MinSphereOfSpheresTraits::D

### ISSUE THREAD

**Issue by mglisse** [CGAL/cgal#2225]
[The documentation of the concept][doc] says:
```C++
    typedef unspecified_type D;
```
as if `D` was a type, but this appears in a section called "Constants" and is used as `static const int D = Traits::D;`, so I assume the documentation of the concept should be fixed.

More importantly, this interface is impossible to use with a dynamic dimension, which looks like a very strong limitation.

  [doc]: http://doc.cgal.org/4.10/Bounding_volumes/classMinSphereOfSpheresTraits.html

**Comment by sgiraudot** [comment]
Inside the code of each model, it is an `int` constant:

```c++
Min_sphere_of_spheres_d_traits_2.h:    static const int D = 2;                  // dimension
Min_sphere_of_spheres_d_traits_3.h:    static const int D = 3;                  // dimension
Min_sphere_of_spheres_d_traits_d.h:    static const int D = Dim_;               // dimension
```

It's only in the separated header files that they are documented as the concept:

```c++
Min_sphere_of_spheres_d_traits_2.h:typedef unspecified_type D; 
Min_sphere_of_spheres_d_traits_3.h:typedef unspecified_type D; 
Min_sphere_of_spheres_d_traits_d.h:typedef unspecified_type D;
```

It seems to me that we should simply replace the `typedef unspecified_type` in both the concept's doc and the three model's doc by `static const int`, as it seems consistent with the code. If we want to keep the implementation opened to dynamic dimension, maybe only document `static int` in the concept?

**Comment by mglisse** [comment]
> It seems to me that we should simply replace the typedef unspecified_type in both the concept's doc and the three model's doc by static const int, as it seems consistent with the code.

That's the minimal fix for consistency.

> If we want to keep the implementation opened to dynamic dimension, maybe only document static int in the concept?

That wouldn't really fit with the way we do things in the rest of CGAL, I'd rather get a consistent interface. But if it is the only (or at least easiest) way to get the functionality, I'll take it.

**Comment by sgiraudot** [comment]
Alright, I made a pull request to integrate the first fix. The dynamic dimension problem probably requires more discussion.

**Comment by mglisse** [comment]
Thanks for the patch. I agree that handling the dynamic dimension case, although it should not be very hard, is not as straightforward as updating the doc to match the code, and it is likely to require a breaking change in the interface.

**Comment by mglisse** [comment]
@lrineau Should I open a separate issue asking for dynamic dimension support if this one is closed?

**Comment by lrineau** [comment]
@mglisse Yes, please. I thought this issue was fixed by #2352, as announced. If there was another issue on that specific topic, that would be clearer than reopening this one.
